library/src/plugins/official/backend/actions/sse.ts

```diff
@@ -30,6 +31,8 @@ const isWrongContent = (err: any) => `${err}`.includes(`text/event-stream`)
 
 export type SSEArgs = {
```

Comment:
The SSEARGS type could be made stricter. Here's an example:

```
export type SSEArgs = {
    method: METHOD
    contentType: 'form'
    selector?: string
    headers?: Record<string, string>
    openWhenHidden?: boolean
} | {
    method: METHOD
    contentType: 'json'
    headers?: Record<string, string>
    openWhenHidden?: boolean
    includeLocal?: boolean
}
```
This way we encode that includeLocal can only be used with contentType: 'json' and selector can only be used with contentType: 'form', since unless I'm mistaken, form submit dont include the signals in any case and selector makes no sense if it's not a form.

Reply:
Good point, but we may want to use `selector` for other content types at some point. I feel like this suggestion, while valid, is unnecessary. Any thoughts, @delaneyj ?

Reply:
I get what you mean, but until we do add a content type that can use selector as well, this makes it clear to anybody reading the code or extending it.

Reply:
> Good point, but we may want to use `selector` for other content types at some point. I feel like this suggestion, while valid, adds unnecessary strictness. Any thoughts, @delaneyj ?

This type constraint is correct. It doesn't mean we can't change it to be more open later. It's a good idea

Reply:
Added.